unittests/soap_over_udp/test_soap_over_udp.cpp

```diff
@@ -0,0 +1,173 @@
+** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+**
+** Contact info@kdab.com if any conditions of this licensing are not
+** clear to you.
+**
+**********************************************************************/
+
+#include "httpserver_p.h"
+#include "KDSoapUdpClient.h"
+#include "KDSoapUdpClient_p.h"
+#include "KDSoapMessage.h"
+#include <QNetworkDatagram>
+#include <QTest>
+#include <QSignalSpy>
+#include "wsdl_wsdd-discovery-200901.h"
+
+Q_DECLARE_METATYPE(QHostAddress)
+
+class SoapOverUdpTest : public QObject
+{
+    Q_OBJECT
+
+private Q_SLOTS:
+    void initTestCase() {
+        qRegisterMetaType<KDSoapMessage>("KDSoapMessage");
+        qRegisterMetaType<KDSoapHeaders>("KDSoapHeaders");
+        qRegisterMetaType<QHostAddress>("QHostAddress");
+    }
+    
+    void testExample() {
+        // This is the example from the KDSoapUdpClient documentation. It is 
+        //   here for compile testing.
+        
+        auto soapUdpClient = new KDSoapUdpClient(this);
+        connect(soapUdpClient, &KDSoapUdpClient::receivedMessage, [=](const KDSoapMessage& message, const KDSoapHeaders& headers, const QHostAddress& address, quint16 port) { 
+            if(message.messageAddressingProperties().action() == QStringLiteral("http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01/ProbeMatches")) {
+                TNS__ProbeMatchesType probeMatches;
+                probeMatches.deserialize(message);
+                qDebug() << "Received probe match from" << address;
+            }
+        });
+        soapUdpClient->bind(3702);
+        
+        TNS__ProbeType probe;
+
+        KDSoapMessage message;
+        message = probe.serialize(QStringLiteral("Probe"));
+        message.setUse(KDSoapMessage::LiteralUse);
+        message.setNamespaceUri(QStringLiteral("http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01"));
+
+        KDSoapMessageAddressingProperties addressing;
+        addressing.setAction(QStringLiteral("http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01/Probe"));
+        addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString(QUuid::WithoutBraces));
+        addressing.setDestination(QStringLiteral("urn:docs-oasis-open-org:ws-dd:ns:discovery:2009:01"));
+        addressing.setReplyEndpointAddress(KDSoapMessageAddressingProperties::predefinedAddressToString(KDSoapMessageAddressingProperties::Anonymous));
+        message.setMessageAddressingProperties(addressing);
+
+        soapUdpClient->sendMessage(message, KDSoapHeaders(), QHostAddress("239.255.255.250"), 3702);
+    }
+    
+    void testSendMessage() {
+        QUdpSocket testSocket;
+        bool rc = testSocket.bind();
+        QVERIFY(rc);
+        
+        KDSoapUdpClient udpClient;
+        
+        udpClient.sendMessage(exampleMessage(), KDSoapHeaders(), QHostAddress::LocalHost, testSocket.localPort());
+        
+        QVERIFY(testSocket.hasPendingDatagrams());
```

Comment:
This test just failed on OSX:

********* Start testing of SoapOverUdpTest *********
Config: Using QtTest library 5.8.0, Qt 5.8.0 (x86_64-little_endian-lp64 shared (dynamic) release build; by Clang 7.0.2 (clang-700.1.81) (Apple))
PASS   : SoapOverUdpTest::initTestCase()
PASS   : SoapOverUdpTest::testExample()
FAIL!  : SoapOverUdpTest::testSendMessage() 'testSocket.hasPendingDatagrams()' returned FALSE. ()
   Loc: [/Users/build/Buildbot/kdsoap-qt5.8-osx2/source/unittests/soap_over_udp/test_soap_over_udp.cpp(94)]
PASS   : SoapOverUdpTest::testReceiveMessage()
PASS   : SoapOverUdpTest::cleanupTestCase()
Totals: 4 passed, 1 failed, 0 skipped, 0 blacklisted, 5ms
********* Finished testing of SoapOverUdpTest *********

Maybe this should use QTRY_VERIFY? I can see how the OS does not guarantee that it's immediately available.

Reply:
Sounds reasonable. This was one of the reasons that I added to verify, to ensure the message has arrived before proceeding.

Reply:
Do you want me to create this change and the change for disabling KDSoapUdpClient for Qt4?

Reply:
I'll test the change and I'll come back to you if it still fails.

The unittest is already disabled for Qt4, and we're dropping Qt4 support after the upcoming release, so that's fine.